search-card.js

```diff
@@ -87,7 +87,10 @@ class SearchCard extends LitElement {
 
     for (var entity_id in this.hass.states) {
       if (entity_id.indexOf(searchText) >= 0) {
-        this.data.push(entity_id);
+        this.data.indexOf(entity_id) === -1 ? this.data.push(entity_id) : null;
```

Comment:
Since the loop iterates over all entities and `this.data` is initialized to an empty list in the beginning of the function, you do not have to check this. It will always be true.